src/components/remote_control/src/resource_allocation_manager_impl.cc

```diff
@@ -162,15 +162,16 @@ void ResourceAllocationManagerImpl::ProcessApplicationPolicyUpdate() {
 RCAppExtensionPtr ResourceAllocationManagerImpl::GetApplicationExtention(
     application_manager::ApplicationSharedPtr application) {
   LOG4CXX_AUTO_TRACE(logger_);
+
+  RCAppExtensionPtr rc_app_extension;
   if (!application) {
-    return NULL;
+    return rc_app_extension;
```

Comment:
Seems kind of dangerous returning an uninitialized pointer... Please initialize rc_app_extension to `NULL` so the pointer does not point to random garbage in memory.

Reply:
@LuxoftAKutsan

Reply:
@JackLivio rc_app_extension is a default constructed shared pointer, so it is already points to NULL, not to random garbage. Also, on bool() operator call for such empty pointer, it will return false as expected. However if you try to init it with NULL explicitly, SDL will crash with DCHECK error so it is much safer to return just an empty shared pointer. For example you could look at FindApp() function in application_manager_impl.cc